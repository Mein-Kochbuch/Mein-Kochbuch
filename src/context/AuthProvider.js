import React, {createContext, useState} from "react";
import axios from "axios";

export const AuthContext = createContext({})

export default function AuthProvider({children}) {

    const [user, setUser] = useState()
    const url = `https://mein-kochbuch.org/api/`

    const login = credentials => {
        return axios
            .post(`${url}/api-token-auth/`, credentials)
            .then(response => response.data)
            .then(data => {
                setUser(data)
            })
    }

    return (
        <AuthContext.Provider value={{user, login}}>
            {children}
        </AuthContext.Provider>
    )
}
