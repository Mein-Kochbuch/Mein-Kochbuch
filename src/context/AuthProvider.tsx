import React, {createContext, ReactElement, useState} from 'react';
import axios from 'axios';
import {LoginResponse} from '../models/Responses';

type User = {
  token: string;
};

interface AuthContextType {
  user?: User;
  login: (credentials: {username: string; password: string}) => Promise<void>;
}

export const AuthContext = createContext<AuthContextType>({
  login: () => Promise.resolve(),
});

export default function AuthProvider({children}: {children: ReactElement}) {
  const [user, setUser] = useState<User>();
  const url = 'https://mein-kochbuch.org/api/';

  const login = (credentials: {username: string; password: string}) => {
    return axios
      .post<LoginResponse>(`${url}api-token-auth/`, credentials)
      .then(response => response.data)
      .then(data => {
        setUser(data);
      });
  };

  return (
    <AuthContext.Provider value={{user, login}}>
      {children}
    </AuthContext.Provider>
  );
}
