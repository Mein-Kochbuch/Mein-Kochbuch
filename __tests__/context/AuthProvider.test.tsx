import 'react-native';
import React from 'react';
import AuthProvider from '../../src/context/AuthProvider';
import renderer from 'react-test-renderer';

describe('AuthProvider', () => {
  test('Snapshot', () => {
    const component = renderer.create(
      <AuthProvider>
        <div>Test children</div>
      </AuthProvider>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
