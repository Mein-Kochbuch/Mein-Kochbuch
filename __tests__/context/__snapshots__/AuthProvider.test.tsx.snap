// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthProvider Snapshot 1`] = `
<div>
  Test children
</div>
`;
